### PR TITLE
Add distance_to_line function and use it for distance_to_polygon

### PR DESCRIPTION
**Pull request by anne-glerum**
Functionality to setup initial conditions, for example temperature based on distance to a ridge.

### CODE REVIEW

File: include/aspect/utilities.h [hunk 1]
```diff
@@ -197,6 +197,18 @@ namespace aspect
     signed_distance_to_polygon(const std::vector<Point<2> > &point_list,
                                const dealii::Point<2> &point);
 
+
+    /**
+    * Given a 2d point and a list of two points that define a line, compute the smallest
```
Comment (bangerth) [inline include/aspect/utilities.h]: Please fix the indentation here.

File: include/aspect/utilities.h [hunk 2]
```diff
@@ -197,6 +197,18 @@ namespace aspect
...
+    * distance of the point to the line segment. Note that when the point's perpendicular
```
Comment (bangerth) [inline include/aspect/utilities.h]: Just drop the "Note that" here -- it's an overused phrase :-)

File: include/aspect/utilities.h [hunk 3]
```diff
@@ -197,6 +197,18 @@ namespace aspect
...
+    * base does not lie on the line segment, the smallest distance to the segment's end
+    * points is calculated.
+    */
+    template <int dim>
+    double
+    distance_to_line(const std::vector<dealii::Point<2> > &point_list,
```
Comment (bangerth) [inline include/aspect/utilities.h]: Since you know that it's only two lines, why not use a `std::array<Point<2>,2>`?

Comment (bangerth) [reply include/aspect/utilities.h]: Also, nothing in this function depends on the template argument `dim` -- I think you can just make it a non-template.

Unless, of course, you want to generalize this to general line segments in 3d as well :-)

File: source/utilities.cc [hunk 4]
```diff
@@ -479,33 +479,62 @@ namespace aspect
...
-            }
+          std::vector<Point<2> > list = {point_list[i], shifted_point_list[i]};
+          distances[i] = distance_to_line<dim>(list, point);
         }
 
       // Return the minimum of the distances of the point to all polygon segments
       return *std::min_element(distances.begin(),distances.end()) * sign;
     }
 
+    template <int dim>
```
Comment (bangerth) [inline source/utilities.cc]: two more empty lines here

**Comment by anne-glerum** [comment]
@gassmoeller My build is not trusted on Jenkins yet, could you start the tester?

**Comment by bangerth** [comment]
It passed three checks, that's probably good enough :-)

File: source/utilities.cc [hunk 5]
```diff
@@ -479,33 +479,61 @@ namespace aspect
...
+       *
+       */
+
+      const unsigned int n_poly_points = point_list.size();
+      AssertThrow(n_poly_points == 2, ExcMessage("A list of points for a line segment should consist of 2 points."));
+
+      // Create vector along the polygon line segment P0 to P1
+      Tensor<1,2> vector_segment = point_list[1] - point_list[0];
+      // Create vector from point P to the second segment point
+      Tensor<1,2> vector_point_segment = point - point_list[0];
```
Comment (bangerth) [inline source/utilities.cc]: make both of these `const`

File: source/utilities.cc [hunk 6]
```diff
@@ -479,33 +479,61 @@ namespace aspect
...
-            }
+          std::array<Point<2>,2 > list = {point_list[i], shifted_point_list[i]};
+          distances[i] = distance_to_line(list, point);
         }
 
       // Return the minimum of the distances of the point to all polygon segments
       return *std::min_element(distances.begin(),distances.end()) * sign;
     }
 
+    double
```
Comment (bangerth) [inline source/utilities.cc]: three empty lines between functions. same after this function

File: source/utilities.cc [hunk 7]
```diff
@@ -479,33 +479,61 @@ namespace aspect
...
-          else if (c2 <= c1)
-            distances[i] = (Tensor<1,2> (shifted_point_list[i] - point)).norm();
-          // perpendicular base line lies on segment
-          else
-            {
-              const Point<2> point_on_segment = point_list[i] + (c1/c2) * vector_segment;
-              distances[i] = (Tensor<1,2> (point - point_on_segment)).norm();
-            }
+          std::array<Point<2>,2 > list = {point_list[i], shifted_point_list[i]};
+          distances[i] = distance_to_line(list, point);
```
Comment (bangerth) [inline source/utilities.cc]: make the `list` variable `const`

File: source/utilities.cc (hunk 6 above)
Comment (anne-glerum) [reply source/utilities.cc]: The number of empty lines between the functions in this file varies, but is usually only 1, therefore I kept it to 1.

Comment (bangerth) [reply source/utilities.cc]: I see. Then let's leave it at that.

**Pull request merged**